### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by maksym-nemera**
[DEMO LINK](https://maksym-nemera.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,57 @@
...
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentTodoId, setCurrentTodoId] = useState<number | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterToDo, setFilterToDo] = useState('all');
```
Comment (silchyn) [inline src/App.tsx:18]:
```suggestion
  const [todoFilter, setTodoFilter] = useState(Filter.All);
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,14 +1,57 @@
...
+    const isActive = filterToDo === 'active';
+    const isCompleted = filterToDo === 'completed';
+    const isAll = filterToDo === 'all';
+
+    const matchesQuery = todo.title.toLowerCase().trim()
+      .includes(query.toLowerCase().trim());
+
+    return (isActive && !todo.completed && matchesQuery)
+      || (isCompleted && todo.completed && matchesQuery)
+      || (isAll && matchesQuery);
```
Comment (silchyn) [inline src/App.tsx:48]: Think about better implementation

File: src/App.tsx [hunk 3]
```diff
@@ -1,14 +1,57 @@
...
+  });
+
+  const getCurrentTodo = (curId: number) => {
+    return visibleTodos.find(todo => todo.id === curId) || null;
+  };
```
Comment (silchyn) [inline src/App.tsx:53]:
```suggestion
  const getCurrentTodo = (id: number) => (
    visibleTodos.find(todo => todo.id === id) || null;
  );
```

File: src/components/TodoFilter/TodoFilter.tsx [hunk 4]
```diff
@@ -1,34 +1,67 @@
...
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            onChange={(event) => {
+              handleOnChange(filterForQuery, event);
+            }}
```
Comment (silchyn) [inline src/components/TodoFilter/TodoFilter.tsx:29]: Put it in a separate callback

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -16,85 +28,70 @@ export const TodoList: React.FC = () => (
...
+                className={cn({
+                  'has-text-success': completed,
+                  'has-text-danger': !completed,
+                })}
+              >
+                {title}
+              </p>
+            </td>
+            <td className="has-text-right is-vcentered">
+
```
Comment (silchyn) [inline src/components/TodoList/TodoList.tsx:74]:
```suggestion
```

**Review by silchyn — CHANGES_REQUESTED** [review]

**Review by silchyn — CHANGES_REQUESTED** [review]
There is a loader when the filter doesn't match any todos, even when nothing is loading

**Review by dinaminahi — APPROVED** [review]
Well done 👍

**Pull request closed without merging**